### DISCUSSION TITLE
Question around popularity requirements, specifically when file extension is not unique

### DISCUSSION THREAD

**Discussion by JakeSteam** [github-linguist/linguist#6741] [Q&A] [answered]
Hey! 

I occasionally use Adobe's "[Lingo](https://en.wikipedia.org/wiki/Lingo_(programming_language))" language, which is almost always contained within an Adobe Director project (so no files). However, if it is exported it becomes a `.ls` file, which understandably gets detected as a LiveScript file ([example repo](https://github.com/n0samu/DirectorCastRipper)), despite Lingo predating LiveScript by a few decades 🙂 

I was interested in working to add a Linguist definition, but due to the "usually embedded in Adobe Director" nature, there are [only ~1.5k results for a quick'n'dirty search for common events](https://github.com/search?q=path%3A*.ls+%28%22on+exitFrame%22+OR+%22on+mouseUp%22+OR+%22on+keyUp%22+OR+%22on+stopMovie%22+OR+%22on+startMovie%22+OR+%22on+idle%22%29&type=code) (misses most files & has a few false positives).

Looking at other PRs, I'm uncertain if Lingo is likely to meet (or be close to meeting) usage requirements. I'm also not sure if this changes for a currently *misidentified* language vs an *undetected* language. 

Any guidance is appreciated, thanks!

**Comment by lildude** [comment, marked as answer]
The requirements don't change for misidentified vs undetected (aka unknown) languages as they both result in the need to add a new language. This is opposed to when the misidentification is between two languages linguist already knows about. Both languages would have already met usage requirements in this case.

One thing to keep in mind is that in the case of shared extensions, we recommend a heuristic which is a regex used to differentiate the languages. You can use this same regex in Search to confirm it works and prove popularity.

**Reply by JakeSteam** [reply]
Gotcha, thanks for the quick response. I'll work on a proper regex and see if it meets requirements!
